model-optimizer/mo/moc_frontend/extractor.py

```diff
@@ -1,35 +1,77 @@
 # Copyright (C) 2018-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-import logging as log
 import re
-from collections import defaultdict
-from copy import copy
-
-import numpy as np
 
+from mo.front.extractor import raise_no_node, raise_node_name_collision
 from mo.utils.error import Error
 
-from ngraph.frontend import InputModel  # pylint: disable=no-name-in-module,import-error
+from ngraph.frontend import InputModel, Place  # pylint: disable=no-name-in-module,import-error
+
+import numpy as np
 
 
 def decode_name_with_port(input_model: InputModel, node_name: str):
     """
     Decode name with optional port specification w/o traversing all the nodes in the graph
-    TODO: in future node_name can specify input/output port groups and indices (58562)
+    TODO: in future node_name can specify input/output port groups as well as indices (58562)
     :param input_model: Input Model
     :param node_name: user provided node name
     :return: decoded place in the graph
     """
-    # Check exact match with one of the names in the graph first
+    found_nodes = []
+    found_node_names = []
+
+    def add_found_node(found_nodes: list, node: Place):
+        """
+        Helper function to add Place node to list of found nodes. Adds only if node is not None
+        :param found_nodes List[Place]: list of found nodes
+        :param node Place: Place node to add to list
+        """
+        found_nodes.append(node) if node else None
+
     node = input_model.get_place_by_tensor_name(node_name)
-    if node:
-        return node
+    found_node_names.append('Tensor:' + node_name)
+    add_found_node(found_nodes, node)
+
+    node = input_model.get_place_by_operation_name(node_name)
+    add_found_node(found_nodes, node)
+    found_node_names.append('Operation:' + node_name)
```

Comment:
Not all frameworks have operation name in a form: 'Operation:' + node_name. The same comment is for tensor, Can we retrieve framework name from input model and raise unsupported exception for `fw != paddle`